### PR TITLE
expose budgets for p2p via cli

### PR DESCRIPTION
**Pull request by stevencartavia**
Hello @emhane !

This is a wip for #11637. I'm trying to follow the same steps as `max-pending-imports` from `crates/node/core/src/args/network.rs`, but the budgets are not showing up in the CLI. I'm uncertain if this is the right approach or if there's a better path to take.

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
are these really meaningful?

I'm skeptical that we should allow configure those via cli

File: crates/node/core/src/args/network.rs [hunk 1]
```diff
@@ -154,6 +160,34 @@ pub struct NetworkArgs {
...
+    /// Default budget to try and drain streams
+    pub budget_try_drain_stream: u32,
+
+    #[arg(long = "budget-try-drain-downloaders", default_value_t = DEFAULT_BUDGET_TRY_DRAIN_DOWNLOADERS, verbatim_doc_comment)]
+    /// Default budget to try and drain headers and bodies download streams.
+    pub budget_try_drain_downloaders: u32,
+
+    #[arg(long = "budget-try-drain-swarm", default_value_t = DEFAULT_BUDGET_TRY_DRAIN_SWARM, verbatim_doc_comment)]
+    /// Budget for draining swarm
+    pub budget_try_drain_swarm: u32,
```
Comment (emhane) [inline crates/node/core/src/args/network.rs:174]:
```suggestion
    /// Budget for draining all ingress traffic, eth [and other capability] messages, and p2p connection updates. When the node has stabilised at max peer count, work done by an iteration in this stream will, to the most part, be accounted for by ingress traffic.
    pub budget_all_ingress: u32,
```

File: crates/node/core/src/args/network.rs [hunk 2]
```diff
@@ -154,6 +160,34 @@ pub struct NetworkArgs {
...
+    /// Budget for draining network handle channel
+    pub budget_try_drain_network_handle_channel: u32,
+
+    #[arg(long = "budget-try-drain-network-transaction-events", default_value_t = DEFAULT_BUDGET_TRY_DRAIN_NETWORK_TRANSACTION_EVENTS, verbatim_doc_comment)]
+    /// Budget for draining network transaction events
+    pub budget_try_drain_network_transaction_events: u32,
+
+    #[arg(long = "budget-try-drain-pending-pool-imports", default_value_t = DEFAULT_BUDGET_TRY_DRAIN_PENDING_POOL_IMPORTS, verbatim_doc_comment)]
+    /// Budget for draining pending pool imports
+    pub budget_try_drain_pending_pool_imports: u32,
```
Comment (emhane) [inline crates/node/core/src/args/network.rs:186]:
```suggestion
    /// Budget for advancing import of transaction batches into pool.
    pub budget_pool_imports: u32,
```

File: crates/node/core/src/args/network.rs [hunk 3]
```diff
@@ -154,6 +160,34 @@ pub struct NetworkArgs {
...
+
+    #[arg(long = "budget-try-drain-pool-imports", default_value_t = DEFAULT_BUDGET_TRY_DRAIN_POOL_IMPORTS, verbatim_doc_comment)]
+    /// Budget for draining pool imports
+    pub budget_try_drain_pool_imports: u32,
```
Comment (emhane) [inline crates/node/core/src/args/network.rs:190]:
```suggestion
    /// Budget for propagating transactions that have successfully been imported into pool.
    pub budget_propagate_gossip: u32,
```

**Review by emhane — CHANGES_REQUESTED** [review]
some more user friendly names and help to start with. good job so far! are you sure you rebuilt the code before running with `--help` in order to see the new flags?

**Comment by stevencartavia** [comment]
> some more user friendly names and help to start with. good job so far! are you sure you rebuilt the code before running with `--help` in order to see the new flags?

yes, with `cargo build`, is there another way I should be building it?

**Comment by stevencartavia** [comment]
@emhane, could you provide any insight into why this check failed? Thanks!

**Comment by emhane** [comment]
> @emhane, could you provide any insight into why this check failed? Thanks!

check that you don't have any other white space other than newlines, in between the flags in the book, i.e. it could be that the cursor isn't at the margin on each new line but in fact some spaces into the page. hope that makes sense.

**Comment by stevencartavia** [comment]
> > @emhane, could you provide any insight into why this check failed? Thanks!
> 
> check that you don't have any other white space other than newlines, in between the flags in the book, i.e. it could be that the cursor isn't at the margin on each empty new line but in fact some spaces into the page. hope that makes sense.

thank you!

**Review by mattsse — COMMENTED** [review]
this isn't useful as cli options

**Pull request closed without merging**